Add date and datetime conversions

swanny85 opened the pull request:
Fulfil requires dates and datetimes to be formatted in an object that contains a iso8601 datetime.  This PR adds a conversion class and integrates it into the model to update the domains automatically.

swanny85 committed:
Create date and datetime conversion to objects

swanny85 committed:
Use conversions in model

swanny85 committed:
Add tests for conversions

swanny85 committed:
Use single quotes in conversion class

swanny85 commented:
I'll fix the reeks.

swanny85 committed:
Reduce complexity in conversions class

swanny85 committed:
Add comments to conversion

swanny85 committed:
Use conversion class method in model

cdmwebs commented:
We'll need this in query, too: https://github.com/knowndecimal/fulfil/blob/main/lib/fulfil/query.rb#L49-L118

cdmwebs commented:
Actually, let's talk through this. We're already doing some of this in another spot.

swanny85 committed:
Create DomainParser

swanny85 committed:
Move conversion methods into fulfil/converter

swanny85 committed:
Remove fulfil/converter

swanny85 committed:
Remove fulfil/conversion references from model

swanny85 committed:
Add domain parser to client

swanny85 commented on lib/fulfil/domain_parser.rb:
```diff
@@ -0,0 +1,64 @@
+    end
+
+    def disable_escape_html_entities
+      return unless defined?(ActiveSupport) && ActiveSupport.respond_to?(:escape_html_entities_in_json=)
```
@cdmwebs `rake test` was erroring out when my guard clause was only `return unless defined?(ActiveSupport)` not sure exactly what to do to fix this codeclimate reek.

swanny85 committed:
Update readme

swanny85 committed:
add method to assign proper converter

swanny85 committed:
Add conversions to query model

swanny85 commented:
Figuring out how to fix these codeclimate reeks.  I've got dates in the queries.  Do those look right @cdmwebs?

swanny85 committed:
Create html entity parser

swanny85 committed:
Include htmlentity parser

swanny85 committed:
Add operator map to fulfil query

swanny85 committed:
Move query to attr_reader

swanny85 committed:
Add note domain parser

swanny85 committed:
Add note to html entity handler

kd-zero committed:
Fix date conversion without global JSON state

Parse domains recursively and only convert Date/DateTime operator values to Fulfil datetime objects. Add regression tests for nested domain parsing and numeric operator comparisons.

kd-zero committed:
Fix rubocop offense in model search

kd-zero committed:
Add RDoc for converter and domain parser
